selfservice/strategy/code/strategy.go

```diff
@@ -383,3 +408,11 @@ const CodeLength = 6
 func GenerateCode() string {
 	return randx.MustString(CodeLength, randx.Numeric)
 }
+
+func maskAddress(input string) string {
+	if strings.Contains(input, "@") {
+		parts := strings.Split(input, "@")
```

Comment:
Use strings.Cut here to split into exactly two parts.

Reply:
Yup, good point. I added some tests to cover all possible (?) variations.